### DISCUSSION TITLE
Research API

### DISCUSSION THREAD

**Discussion by Crazycolbster** [OpenRCT2/OpenRCT2#18792] [Feature requests] [answered]
It’d be cool if there were an API that could let us manipulate things related to research. For instance, unlocking rides, changing the order of the tech list, setting the progress of research, etc. Any chance of something like this being made? Thanks!

**Comment by MASHzj** [comment]
You can do this via Cheats > object selection and Cheats > inventions list. Or is that not what you mean?

**Reply by Crazycolbster** [reply]
Specifically, I’d like to use this hypothetical API in writing an extension to make OpenRCT2 compatible with the Archipelago randomizer system. This would require automated calls to change/shuffle the list, identify what’s in it, etc. Basically, I’d want to be able to do what you said in code instead of manually

**Comment by Crazycolbster** [comment, marked as answer]
They added it! Thanks @IntelOrca!
